### PR TITLE
Fix xfidruneread for small reads and multi-byte characters

### LINKED ISSUE #255: Crash while reading from xdata
This is very much reproducible: ~~read from `xdata` with negative `q0`~~. (see below)

```
panic: runtime error: slice bounds out of range [-2:]

goroutine 51 [running]:
panic(0x634ce0, 0xc000016960)
        /home/big/go/src/runtime/panic.go:706 +0x2c2 fp=0xc0001f7dd0 sp=0xc0001f7d40 pc=0x42ea42
runtime.goPanicSliceB(0xfffffffffffffffe, 0x2)
        /home/big/go/src/runtime/panic.go:87 +0xa3 fp=0xc0001f7e18 sp=0xc0001f7dd0 pc=0x42d543
main.xfidruneread(0xc0000c2000, 0xc00011a118, 0xcb0, 0xcb2, 0xc0001f7f30)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.1-0.20190628123847-c80f7ecdf670/xfid.go:944 +0x397 fp=0xc0001f7ec8 sp=0xc0001f7e18 pc=0x5d43d7
main.xfidread(0xc0000c2000)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.1-0.20190628123847-c80f7ecdf670/xfid.go:372 +0x9ba fp=0xc0001f7f88 sp=0xc0001f7ec8 pc=0x5d117a
main.xfidctl(0xc0000c2000, 0x69f500, 0xc000010120)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.1-0.20190628123847-c80f7ecdf670/xfid.go:52 +0x9c fp=0xc0001f7fc8 sp=0xc0001f7f88 pc=0x5cf68c
runtime.goexit()
        /home/big/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc0001f7fd0 sp=0xc0001f7fc8 pc=0x45bff1
created by main.xfidallocthread
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.1-0.20190628123847-c80f7ecdf670/acme.go:619 +0x1d4
```

### PR DESCRIPTION
**Pull request by fhs**
We were crashing if we were trying to read less than 4 (`utf8.UTFMax`)
runes. The function also didn't handle muti-byte characters correctly
and it returned the number of bytes read instead of the number of runes.

I've changed the function to be more similar to the C code, decoding
runes from the beginning of the bytes buffer instead of try to find the
rune boundary from the end of the buffer. Finding the boundary doesn't
solve the problem of computing the number of runes, which we need to
return.

Fixes #255

### CODE REVIEW

File: xfid.go [hunk 1]
```diff
@@ -925,46 +925,37 @@ func xfidutfread(x *Xfid, t *Text, q1 int, qid int) {
...
-		}
-	}
-	// add all further full runes
-	for len(buf[i:]) > 0 {
-		ru, count := utf8.DecodeRune(buf[i:])
-		if ru == utf8.RuneError {
-			break
-		} else {
-			i += count
+
```
Comment (rjkroege) [inline xfid.go]: Could you maybe add a method comment explaining what xfidruneread is really suppose to do given that you have the best understanding of this code at present? 😊

File: xfid.go [hunk 2]
```diff
@@ -925,46 +925,37 @@ func xfidutfread(x *Xfid, t *Text, q1 int, qid int) {
...
+	m := len(buf)
+	if len(buf) > int(x.fcall.Count) {
```
Comment (rjkroege) [inline xfid.go]: Per above: conceivably I'm not understanding this code. But,  it seems like our goal is to end up with  `buf` that contains at most `x.fcall.Count` bytes so long as `buf` is a complete valid UTF8 byte sequence -- in particular no partial rune encodings?

File: xfid.go (hunk 1 above)
Comment (fhs) [reply xfid.go]: Will do. :+1:

File: xfid.go (hunk 2 above)
Comment (fhs) [reply xfid.go]: That's exactly right. I'll add a method doc.

**Pull request merged**